From 889bcc393f15f2b260457a4b92d74a348025c65f Mon Sep 17 00:00:00 2001
From: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
Date: Tue, 21 Jul 2026 19:02:50 +0000
Subject: [PATCH] dt-bindings: net: airoha: make xPON Ethernet phandle optional

Allow the xPON MAC driver to discover the EN7523 GDM2 netdev when
the Ethernet datapath phandle is omitted. Keep the phandle as an
optional compatibility override.

Signed-off-by: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
---
 .../devicetree/bindings/net/airoha,en7523-xpon.yaml          | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Documentation/devicetree/bindings/net/airoha,en7523-xpon.yaml b/Documentation/devicetree/bindings/net/airoha,en7523-xpon.yaml
index 0a82e9419..fa6366fad 100644
--- a/Documentation/devicetree/bindings/net/airoha,en7523-xpon.yaml
+++ b/Documentation/devicetree/bindings/net/airoha,en7523-xpon.yaml
@@ -53,7 +53,8 @@ properties:
   ethernet:
     $ref: /schemas/types.yaml#/definitions/phandle
     description:
-      Phandle to the airoha,eth-mac GDM2 child used as the xPON datapath.
+      Optional phandle to the airoha,eth-mac GDM2 child used as the xPON
+      datapath. When omitted, the driver discovers the EN7523 GDM2 netdev.
 
   airoha,scu:
     $ref: /schemas/types.yaml#/definitions/phandle
@@ -100,7 +101,6 @@ required:
   - compatible
   - reg
   - interrupts
-  - ethernet
   - airoha,scu
   - sfp
 
@@ -137,7 +137,6 @@ examples:
         reg-names = "mac";
         interrupts = <GIC_SPI 42 IRQ_TYPE_LEVEL_HIGH>;
         interrupt-names = "mac";
-        ethernet = <&gdm2>;
         airoha,scu = <&scuclk>;
         airoha,pon-mode = "gpon";
         airoha,gpon-serial-number = [49 4e 56 50 71 43 86 02];
-- 
2.55.0

